### PR TITLE
Fix writing functions of time in observer, bind Domain items in Python

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

- Functions of time were not written by evolution executables because they are stored in the _mutable_ global cache. This is now fixed, and I updated the test volume H5 file in the repo with output from the current EvolveScalarWave3D executable.
- Bind Domain and FunctionsOfTime in Python so they can be read from H5 volume files.
- Add an `inertial_coordinates` function in Python that uses the Domain and FunctionsOfTime to map logical to inertial coords. This will allow things like interpolation routines in Python, and plotting the Lagrange interpolation in DG elements.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Domain/Python/InertialCoordinates.cpp [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        const tnsr::I<DataVector, Dim, Frame::ElementLogical> logical_coords(
+            element_logical_coords);
+        tnsr::I<DataVector, Dim, Frame::Inertial> inertial_coords{};
+
+        // Map logical to inertial coords
+        const auto& block = domain.blocks()[element_id.block_id()];
+        if (block.is_time_dependent()) {
+          if (not time.has_value()) {
+            throw std::runtime_error(
+                "The 'time' argument is required for time-dependent domains.");
```
Comment (knelli2) [inline src/Domain/Python/InertialCoordinates.cpp]: Do we have a way to test errors in python? The two `throw` statements don't seem to be tested.

File: tests/Unit/Domain/Python/Test_InertialCoordinates.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            inertial_coordinates_3d(element_logical_coords,
+                                    element_id=element_id,
+                                    domain=domain,
+                                    time=0.,
+                                    functions_of_time=functions_of_time))
+
+        # Domain is [0, 2 pi]^3
+        npt.assert_allclose(
+            inertial_coords,
+            np.array([[0., np.pi], [0., 2 * np.pi], [0., 2 * np.pi]]))
```
Comment (knelli2) [inline tests/Unit/Domain/Python/Test_InertialCoordinates.py]: The comment and the actual array don't match.

**Review by knelli2 — CHANGES_REQUESTED** [review]

File: tests/Unit/Domain/Python/Test_InertialCoordinates.py (hunk 2 above)
Comment (nilsvu) [reply tests/Unit/Domain/Python/Test_InertialCoordinates.py]: It's correct, extended it a bit to make it clearer :)

File: src/Domain/Python/InertialCoordinates.cpp (hunk 1 above)
Comment (nilsvu) [reply src/Domain/Python/InertialCoordinates.cpp]: Done, also changed to `invalid_argument` error which maps to `ValueError` in Python

**Comment by nilsvu** [comment]
Rebased on #4543

**Review by knelli2 — COMMENTED** [review]
Looks good. You can squash.

**Comment by nilsvu** [comment]
Thanks for reviewing!

**Review by knelli2 — APPROVED** [review]

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**